tests/integration/cluster_setup_test.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/rook/rook/pkg/model"
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/enums"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestClusterSetupSuite(t *testing.T) {
```

Comment:
Can we include this test with another e2e test? As a philosophy i think our e2e tests need to stay as integrated as possible, unlike unit tests which are all independent. For example, the smoke test could run with bluestore and the helm tests could run with filestore. This way we don't need to add any more time to the tests. If it's a new suite like this, we are adding several minutes to the already painful length of the tests.

Reply:
These new tests will only run as part of full regression, not as part of the smoke suite.  I really do like having an explicit suite here to fully verify cluster setup succeeded, as opposed to implicitly checking that filestore/bluestore are both working indirectly via other tests.

I think a new suite here is OK since it is not part of the smoke test that runs on PR branches.  Full regression tests only run on master, right?

Reply:
actually that's only if you set `[smoke only]` in the PR. By default all the tests run

Reply:
ah, you are right and as you also pointed out to me, my new suite added 6 minutes to the integration test run.  That's clearly not desirable, so I am now relying on the multi cluster test suite that sets up 2 clusters to verify that they are both healthy.  One of them is using bluestore and one of them is using filestore.

I have also updated the `checkIfRookClusterIsInstalled` function to verify cluster health in addition to simply verifying that the Rook pods are running.